### DISCUSSION TITLE
Trying to wrap my head around sessions / workspaces

### DISCUSSION THREAD

**Discussion by sommerper** [wezterm/wezterm#4741] [Q&A]
### What Operating System(s) are you running on?

Linux X11

### Which Wayland compositor or X11 Window manager(s) are you using?

popos cosmic

### WezTerm version

wezterm 20230712-072601-f4abf8fd

### Ask your question!

I'm trying to find out if I can replicate my tmux setup with wezterm alone and even after reading the docs and the questions here I still don't know what I'm supposed to do.

I have this script on my local machine and several servers that I connect to:
```wd () {
    WORKDIR=$(fd . $HOME/development --maxdepth 4 --type d --exclude '*node_modules*' --exclude '*/.*' | fzf | sed 's:/*$::') || return
    if [ -z "$WORKDIR" ]
    then
        return
    fi
    TITLE=${WORKDIR##*/}
    echo "\x1b]1337;SetUserVar=panetitle=$(echo -n $TITLE | base64)\x07"
    tmux new-session -A -s ${WORKDIR} -c ${WORKDIR}
    echo "\x1b]1337;SetUserVar=panetitle=$(echo -n "" | base64)\x07"
}

...

if [[ $TMUX_PANE ]]; then
    # check if $HOME/tmux_history/ exists
    if [ ! -d "$HOME/tmux_history" ]; then
        mkdir $HOME/tmux_history
    fi
    tmux_session_name=$(tmux display-messag -p '#S' | sed 'y#/#_#' | cut -c2-)
    HISTFILE=$HOME/tmux_history/.history_${tmux_session_name}
    alias dt='tmux detach'

```

Basically what the function does is go to start a session or connect to an existing one and then once it's opened I set the history file to that session alone.

From what I can read I think I'm supposed to:
```
wezterm cli spawn --new-window --workspace $TITLE
```

Which seems to do something because I can see it when running:
```
wezterm cli list
```
which gives me:
```
    0    20     20 default        172x44 wezterm
    1    21     21 config_wezterm 80x24  ~
```

But I don't know how to activate it so I can start working in that session let alone how to save and restore (like with tmux ressurect).

Am I missing something or am I just stupid and misunderstanding what sessions / workspaces is all about?

Thanks for any help!
